Composite/Plugins/Functions/WidgetFunctionProviders/StandardWidgetFunctionProvider/DataReference/PageReferenceSelectorWidgetFunctionBase.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Xml.Linq;
+using Composite.Core.Types;
+using Composite.Data.Types;
+using Composite.Functions;
+using Composite.Plugins.Elements.ElementProviders.PageElementProvider;
+using Composite.Plugins.Functions.WidgetFunctionProviders.StandardWidgetFunctionProvider.Foundation;
+
+namespace Composite.Plugins.Functions.WidgetFunctionProviders.StandardWidgetFunctionProvider.DataReference
+{
+    internal abstract class PageReferenceSelectorWidgetFunctionBase : CompositeWidgetFunctionBase
+    {
+        protected const string CompositeNameBase = CommonNamespace + ".DataReference";
+
+        protected PageReferenceSelectorWidgetFunctionBase(string compositeName, Type returnType, EntityTokenFactory entityTokenFactory) : base(compositeName, returnType, entityTokenFactory)
+        {
+            base.AddParameterProfile(
+                new ParameterProfile("HomePageIdFilter", typeof(Guid?), false,
```

Comment:
Probably "HomePage" or "HomePageId" will be better, but i'm not sure, yoo can leave as is.

Reply:
changed to HomePageId, to be consistent